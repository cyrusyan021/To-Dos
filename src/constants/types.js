const BUTTON_TYPE = {
  clear: 'clear',
  outlined: 'outlined',
  warning: 'warning',
  icon: 'icon',
  primary: 'primary',
}

const SIZE = {
  sm: 'sm',
  md: 'md',
  lg: 'lg',
}

/* --- props --- */
const BUTTON_TYPE_VALID_PROPS = Object.values(BUTTON_TYPE);

const SIZE_VALID_PROPS = Object.values(SIZE);

export {
  BUTTON_TYPE,
  BUTTON_TYPE_VALID_PROPS,
  SIZE,
  SIZE_VALID_PROPS,
}