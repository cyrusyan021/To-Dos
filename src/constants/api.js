const API_ERROR_MESSAGE = 'Something went wrong. Please try again later.';

export {
  API_ERROR_MESSAGE,
}