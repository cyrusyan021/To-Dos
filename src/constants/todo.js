const ACTIVE_STATUS = {
  all: 'all',
  active: 'active',
  completed: 'completed',
}

const ACTIVE_STATUS_OPTIONS = [
  { id: ACTIVE_STATUS.all, name: 'All' },
  { id: ACTIVE_STATUS.active, name: 'Active' },
  { id: ACTIVE_STATUS.completed, name: 'Completed' },
];

export {
  ACTIVE_STATUS,
  ACTIVE_STATUS_OPTIONS,
}