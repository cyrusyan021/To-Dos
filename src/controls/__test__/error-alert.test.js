import React from "react"
import { shallow } from 'enzyme';

import ErrorAlert from 'controls/error-alert';

import { API_ERROR_MESSAGE } from 'constants/api';

describe("Error Alert", () => {
  const props = {
    error: 'Testing Error Message',
  };

  it("renders correctly with default error message", () => {
    const wrapper = shallow(<ErrorAlert />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').first().text()).toEqual(API_ERROR_MESSAGE);
  });

  it("renders correctly with specified error message", () => {
    const wrapper = shallow(<ErrorAlert { ...props } />);
    expect(wrapper.find('div').first().text()).toEqual(props.error);
  });
});