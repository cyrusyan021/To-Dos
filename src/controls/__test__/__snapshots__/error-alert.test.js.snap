// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Alert renders correctly with default error message 1`] = `ShallowWrapper {}`;
